scripts/pylib/twister/twisterlib/handlers.py

```diff
@@ -706,6 +707,23 @@ def _get_serial_device(self, serial_pty, hardware_serial):
 
         return serial_device, ser_pty_process
 
+    def _create_serial_pty_script(self, runner):
```

Comment:
ok, so you are creating the pty script on the fly, if I understand this correctly, it is an abuse of the `--device-serial-pty` option, we probably need something else to cover this case.

Reply:
"abuse" is probably a strong term.

I prefer "reuse" - we're simply reusing west's rtt integration.

Reply:
```
    device.add_argument("--device-serial-pty",
                        help="""Script for controlling pseudoterminal.
                        Twister believes that it interacts with a terminal
                        when it actually interacts with the script.

                        E.g "twister --device-testing
                        --device-serial-pty <script>
```

Not using a script here is wrong usage. If you want to use a built-in rtt support you are adding, you need to find another way, maybe yet another option that deals with `keywords`